include/bmi/Bmi_Py_Adapter.hpp

```diff
@@ -236,24 +236,31 @@ namespace models {
                  * this might cause trouble on certain systems, since (depending on the particular sizes of types) that
                  * could produce duplicate "case" values.
                  */
-                //TODO: include other numpy type strings, https://numpy.org/doc/stable/user/basics.types.html
-                if ( (py_type_name == "int" || py_type_name == "int16") && item_size == sizeof(short)) {
+                if (item_size == sizeof(signed char) && (py_type_name == "int8" || py_type_name == "numpy.int8")) {
+                    return "signed char";
+                } else if (item_size == sizeof(unsigned char) && (py_type_name == "uint8" || py_type_name == "numpy.uint8")) {
+                    return "unsigned char";
+                } else if (item_size == sizeof(short) && (py_type_name == "int" || py_type_name == "int16" || py_type_name == "numpy.int16")) {
                     return "short";
-                } else if ( (py_type_name == "int" || py_type_name == "int32" )&& item_size == sizeof(int)) {
+                } else if (item_size == sizeof(unsigned short) && (py_type_name == "uint16" || py_type_name == "numpy.uint16" || py_type_name == "ushort" || py_type_name == "numpy.ushort")) {
+                    return "unsigned short";
+                } else if (item_size == sizeof(int) && (py_type_name == "int" || py_type_name == "int32" || py_type_name == "numpy.int32" || py_type_name == "intc" || py_type_name == "numpy.intc")) {
```

Comment:
`numpy.(u)long` is a possible match for the 32-bit integer cases, as well as the 64-bit cases